### PR TITLE
Only get resource when the class is an identifier

### PR DESCRIPTION
**Pull request by charleswaddell**
https://hippoeducation.tpondemand.com/entity/627

### CODE REVIEW

**Comment by gauthierm** [comment]
Does `ToOneRelationship::encodeIdentifier` need to be updated?

**Comment by charleswaddell** [comment]
> Does `ToOneRelationship::encodeIdentifier` need to be updated?

Why would it need to be updated? **Edit** It does not need to be updated. It doesn't load the resource.

**Comment by gauthierm** [comment]
> Why would it need to be updated?

It uses `if ($resource instanceof ResourceIdentifier) {...}`

**Review by gauthierm — CHANGES_REQUESTED** [review]
https://github.com/charleswaddell/json-api-client/pull/3

**Review by gauthierm — APPROVED** [review]

File: src/AbstractResource.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    {
+        $this->checkStore();
+
+        return $this->store->find($this->getType(), $this->getId());
+    }
+
+    // }}}
+    // {{{ public function encodeIdentifier()
+
+    public function encodeIdentifier()
```
Comment (gauthierm) [inline src/AbstractResource.php]: needs options = []

File: src/AbstractResource.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                'id' => $this->getId(),
+                'type' => $this->getType(),
+            ]
+        ];
+    }
+
+    // }}}
+    // {{{ abstract public function encode()
+
+    abstract public function encode();
```
Comment (gauthierm) [inline src/AbstractResource.php]: needs array optinos = []

**Review by gauthierm — CHANGES_REQUESTED** [review]

File: src/AbstractResource.php (hunk 2 above)
Comment (charleswaddell) [reply src/AbstractResource.php]: Optinos sound delicious.

**Review by gauthierm — APPROVED** [review]

**Pull request merged**